tools/file_impact.py

```diff
@@ -0,0 +1,48 @@
+from __future__ import print_function
+import sys
+import argparse
+import json
+from os.path import abspath, join, dirname
+
+# Be sure that the tools directory is in the search path
+ROOT = abspath(join(dirname(__file__), ".."))
+sys.path.insert(0, ROOT)
+
+from tools.build_api import prepare_toolchain, scan_resources
+from tools.toolchains import mbedToolchain, TOOLCHAIN_CLASSES, TOOLCHAIN_PATHS
+from tools.targets import TARGET_MAP
+from tools.config import ConfigException
+from tools.arm_pack_manager import do_queue, Reader
+
+def should_be_built(target_name, toolchain_name, to_find):
+    if toolchain_name not in TARGET_MAP[target_name].supported_toolchains:
+        return False
+    k = prepare_toolchain(ROOT, "BUILD", target_name, toolchain_name,
+                            notify=lambda _, __: True)
+    try:
+        r = scan_resources([ROOT], k)
+        combo_files = set(
+            r.headers + r.c_sources + r.cpp_sources + r.s_sources +
+            r.objects + r.libraries + r.hex_files + r.bin_files +
+            r.json_files)
```

Comment:
What's included in `json_files` ? Are `targets.json` and compilation profiles included ?

Reply:
Anything that ends with `.json` in a scan is added. This includes `targets.json` and any `mbed_lib.json` files. This does not include the build profiles, as there is a `.mbedignore` containing:
```
*
```
in the root of the tools directory.

Reply:
Thanks for the clarification. Would it be a good thing to ensure everything continue to work if something is changed in the `tools` directory ?

Reply:
For this function they will ignore them. If you look elsewhere in the script, any unknown files cause every target to be emitted.